From 0b13c3038f06666333654e0adfea274a396a5b6d Mon Sep 17 00:00:00 2001
From: "Lucas Perais (lpe)" <lpe@odoo.com>
Date: Tue, 30 Jul 2019 15:35:14 +0000
Subject: [PATCH] [FIX] l10n_be_intrastat: propagate country_id

Before this commit, the field intrastat_country_id was not propagated when refunding an invoice

After this commit, the refund invoice takes the same values for that field
as the invoice it is refunding

OPW 2028333

closes odoo/odoo#35309

Signed-off-by: Lucas Perais (lpe) <lpe@odoo.com>
---
 addons/l10n_be_intrastat/models/l10n_be_intrastat.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/addons/l10n_be_intrastat/models/l10n_be_intrastat.py b/addons/l10n_be_intrastat/models/l10n_be_intrastat.py
index beff3b4..99f9945 100644
--- a/addons/l10n_be_intrastat/models/l10n_be_intrastat.py
+++ b/addons/l10n_be_intrastat/models/l10n_be_intrastat.py
@@ -22,6 +22,17 @@ class AccountInvoice(models.Model):
         help='Intrastat country, delivery for sales, origin for purchases',
         domain=[('intrastat', '=', True)])
 
+    @api.model
+    def _prepare_refund(self, invoice, date_invoice=None, date=None, description=None, journal_id=None):
+        """
+        Correctly set intrastat_country_id
+        @override
+        """
+        values = super(AccountInvoice, self)._prepare_refund(invoice, date_invoice, date, description, journal_id)
+        if 'intrastat_country_id' not in values:
+            values['intrastat_country_id'] = invoice.intrastat_country_id.id
+        return values
+
 
 class IntrastatRegion(models.Model):
     _name = 'l10n_be_intrastat.region'
-- 
2.7.4

